File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        
        
            unordered_map<int,int>gzip;
            for(int i=0;i<nums.size();i++){        
                        if(gzip.find(nums[i])!=gzip.end()){
                                if(i-gzip[nums[i]]<=k)
                                            return true;                            
                        }
                        gzip[nums[i]]=i;   
            }
        
            return false;
        
    }
};

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        
        
        dict1={}
        
        for indx,num in enumerate(nums):
            
            if num not in dict1:        
                    dict1[num]=indx
                    
            else:
                    print(indx,dict1[num])
                    if indx-dict1[num]<=k:
                        return True
                    dict1[num]=indx
            
            
        return False